supervisor/homeassistant/module.py

```diff
@@ -335,16 +335,21 @@ async def backup(self, tar_file: tarfile.TarFile) -> None:
 
             # Backup data config folder
             def _write_tarfile():
-                with tar_file as backup:
-                    # Backup metadata
-                    backup.add(temp, arcname=".")
-
-                    # Backup data
-                    atomic_contents_add(
-                        backup,
-                        self.sys_config.path_homeassistant,
-                        excludes=HOMEASSISTANT_BACKUP_EXCLUDE,
-                        arcname="data",
+                try:
+                    with tar_file as backup:
+                        # Backup metadata
+                        backup.add(temp, arcname=".")
+
+                        # Backup data
+                        atomic_contents_add(
+                            backup,
+                            self.sys_config.path_homeassistant,
+                            excludes=HOMEASSISTANT_BACKUP_EXCLUDE,
+                            arcname="data",
+                        )
+                except (tarfile.TarError, OSError) as err:
+                    _LOGGER.warning(
```

Comment:
This needs to raise `BackupError`

Reply:
Please handle this at line 350/355

Reply:
And raise of curse an error, you can create an HomeAssistantBackupError which is inherit from BackupError and HomeAssistantError

Reply:
I tried to replicate what we did previously, where that folder has been handled by the generic folder store code:

https://github.com/home-assistant/supervisor/blob/main/supervisor/backups/backup.py#L404

Should we change it there too then?

Reply:
yes, we can't "hide" issues with backups by just logging it, we need to raise and cancel